src/WorkspaceDependencyViewExtension/WorkspaceDependencyView.xaml.cs

```diff
@@ -145,6 +146,40 @@ private void DownloadPackage(object sender, RoutedEventArgs e)
             packageInstaller.DownloadAndInstallPackage(package);
             DependencyRegen(currentWorkspace);
         }
+
+        /// <summary>
+        /// Handler of button which user click when choosing to keep the
+        /// installed version of package instead of the specified one.
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void KeepLocalPackage(object sender, RoutedEventArgs e)
+        {
+            try
+            {
+                var info = ((PackageDependencyRow)((Button)sender).DataContext).DependencyInfo;
+                UpdateWorkspaceToUseInstalledPackage(info);
+            }
+            catch(Exception) {}
+        }
+
+        /// <summary>
+        /// Update current workspace in memory to keep the installed package
+        /// instead of keep referencing the dependency info saved in DYN
+        /// </summary>
+        /// <param name="info">Target PackageDependencyInfo to update version</param>
+        internal void UpdateWorkspaceToUseInstalledPackage(PackageDependencyInfo info)
+        {
+            var pmExtension = dependencyViewExtension.ViewLoadedParams.ExtensionManager.Extensions.Where(x => x.Name == "DynamoPackageManager").FirstOrDefault() as PackageManagerExtension;
```

Comment:
you can likely make this safer and shorter by using `OfType<PackageManagerExtension>()`